pyccel/parser/semantic.py

```diff
@@ -791,29 +789,25 @@ def _visit_CommentBlock(self, expr, **settings):
         return expr
     def _visit_AnnotatedComment(self, expr, **settings):
         return expr
-    def _visit_Integer(self, expr, **settings):
-        if isinstance(expr, Integer):
-            return expr
-        elif isinstance(expr, sp_Integer):
-            return Integer(expr)
-        else:
-            raise TypeError("Integer type is not sympy Integer or pyccel Integer")
-    def _visit_Float(self, expr, **settings):
-        if isinstance(expr, Float):
-            return expr
-        elif isinstance(expr, sp_Float):
-            return Float(expr)
-        else:
-            raise TypeError("Float type is not sympy Float or pyccel Float")
-    def _visit_Complex(self, expr, **settings):
+    def _visit_LiteralInteger(self, expr, **settings):
+        return LiteralInteger(expr)
```

Comment:
Shouldn't we return just `expr` here?
Actually, it sounds like we only need one `_visit_Literal` function to cover all cases.

Reply:
Ah! This is how I managed to miss the cast in `_visit_Integer`. I put it in the wrong function.
I agree, I think there is a lot of overlap between the literal functions (there could be more if we didn't inherit from sympy.Integer etc)